bears/general/SpaceConsistencyBear.py

```diff
@@ -38,7 +41,48 @@ def run(self,
         result_texts = []
         additional_info_texts = []
 
-        for line_number, line in enumerate(file, start=1):
+        def get_blanklines_nr_start():
+            line_nr_start = False
+            rev_enumerated_zip_obj = zip(reversed(range(1, len(file) + 1)),
+                                         reversed(file))
+            rev_enumerated_tuple = tuple(rev_enumerated_zip_obj)
```

Comment:
Explain the reason for converting it into a tuple. You are losing the efficiency of a zip/generator like object by converting it into a list like object

Reply:
because zip return an iterator, so after zip return an iterator of tuples of `line and line no.` then to iterate over that iterator in `for loop` 👇 I converted it to a tuple.

Reply:
Can you a little bit explain how I'm losing the efficiency of zip by converting it to a tuple?

Reply:
I think you can iterate normally in a for loop over an iterator, no need to convert it to tuple. Plus using iterators/generator like objects is highly efficient

Reply:
>Can you a little bit explain how I'm losing the efficiency of zip by converting it to a tuple?

 list comprehension creates the entire list in memory first while the generator expression will create the items on the fly, so you are able to use it for very large (and also infinite!) sequences.

Reply:
so if you only have to loop once then generator like objects are considered far more efficient and faster than list like objects. I think you can read the entire explanation on stackoverflow. there are many questions regarding this

Reply:
I think I've tried this when I was making changes but that didn't work.
btw, let me try this again.May be I was making some mistake ;)

Reply:
it working 👍 let see it won't error in CI

Reply:
thanks for your explanation :) @ishanSrt